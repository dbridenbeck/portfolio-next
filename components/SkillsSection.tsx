import styled from "styled-components";
import Pill from "./SharedComponents/Pill";
import { devices } from "../utils/cssBreakpoints";

const SkillsContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 100%;
  @media ${devices.mobileLandscape} {
    font-size: 1.125em;
  }
  @media ${devices.largeScreen} {
    font-size: 1.25em;
  }
`;

const Headline = styled.h2`
  font-size: 1.75em;
  font-weight: 600;
  margin: 0;
  padding: 0;
`;

const Info = styled.p`
  font-size: 0.875em;
  line-height: 1.5em;
  margin: 0;
  padding: 0;
`;

const TechContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  font-size: 0.825em;
`;

const Styledh3 = styled.h3`
  margin: 0.5em 0;
  padding: 0;
  font-size: 0.825em;
`;

const languagesAndTools = [
  "JavaScript ES6",
  "TypeScript",
  "NodeJS",
  "Python",
  "React",
  "Redux",
  "Gatsby.js",
  "Next.js",
  "Styled-Components",
  "Webpack",
  "Express",
  "RESTful API's",
  "MongoDB",
  "Mongoose",
  "Socket.io",
  "Jest",
  "Github",
  "TDD",
  "Figma",
  "Adobe Creative Suite",
  "Shopify",
];

const SkillsSection: React.FC = () => (
  <SkillsContainer>
    <Headline>I use code to bring people together and have fun.</Headline>
    <Info>
      I am a detail oriented and intrinsically motivated software engineer with
      five years’ experience creating custom websites using HTML/CSS/JavaScript,
      and eight years’ experience working directly with clients to solve complex
      technical challenges.
    </Info>
    <div>
      <Styledh3>Languages and Tools:</Styledh3>
      <TechContainer>
        {languagesAndTools.map((techName) => (
          <Pill key={techName} setMargin={"0.25em"}>
            {techName}
          </Pill>
        ))}
      </TechContainer>
    </div>
  </SkillsContainer>
);

export default SkillsSection;
