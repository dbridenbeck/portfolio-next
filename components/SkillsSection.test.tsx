import React from "react";
import { render } from "@testing-library/react";
import SkillsSection from "./SkillsSection";

describe("SkillsSection", () => {
  it("should render without errors", () => {
    const { container } = render(<SkillsSection />);
    expect(container).toMatchSnapshot();
  });
});
