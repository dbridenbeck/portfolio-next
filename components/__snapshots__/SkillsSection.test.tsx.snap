// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsSection should render without errors 1`] = `
<div>
  <div
    class="SkillsSection__SkillsContainer-su0qe3-0 fgNfwp"
  >
    <h2
      class="SkillsSection__Headline-su0qe3-1 bZAzkz"
    >
      I use code to bring people together and have fun.
    </h2>
    <p
      class="SkillsSection__Info-su0qe3-2 hsNCjH"
    >
      I am a detail oriented and intrinsically motivated software engineer with five years’ experience creating custom websites using HTML/CSS/JavaScript, and eight years’ experience working directly with clients to solve complex technical challenges.
    </p>
    <div>
      <h3
        class="SkillsSection__Styledh3-su0qe3-4 dKEcvO"
      >
        Languages and Tools:
      </h3>
      <div
        class="SkillsSection__TechContainer-su0qe3-3 esTUjr"
      >
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          JavaScript ES6
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          TypeScript
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          NodeJS
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Python
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          React
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Redux
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Gatsby.js
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Next.js
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Styled-Components
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Webpack
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Express
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          RESTful API's
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          MongoDB
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Mongoose
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Socket.io
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Jest
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Github
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          TDD
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Figma
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Adobe Creative Suite
        </div>
        <div
          class="Pill__PillContainer-c9peiv-0 dSkIQ"
        >
          Shopify
        </div>
      </div>
    </div>
  </div>
</div>
`;
